### PR TITLE
Clients: Don't disable file registration when uploading to non-deterministic RSE; Fix #2620

### LINKED ISSUE #2620: Upload fails to register file if pfn provided and non-deterministic RSE
Motivation
----------

If the RSE is non-deterministic and a pfn is provided then the client prints `Upload with given pfn implies that no_register is True, except non-deterministic RSEs` and fails to register the replica.

### PR DESCRIPTION
**Pull request by illingwo**
Clients: Don't disable file registration when uploading to non-deterministic RSE; Fix #2620

### CODE REVIEW

File: lib/rucio/client/uploadclient.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def upload(self, items, summary_file_path=None):
             # if register_after_upload, file should be overwritten if it is not registered
             # otherwise if file already exists on RSE we're done
             if register_after_upload:
-                if rsemgr.exists(rse_settings, file_did):
+                if rsemgr.exists(rse_settings, pfn if pfn else file_did):
```
Comment (TomasJavurek) [inline lib/rucio/client/uploadclient.py:169]: Why this change? pfn is preferred, otherwise rsemgr needs to extract pfn itself.

**Comment by bari12** [comment]
Implemented in @TomasJavurek PR
